src/scripts/classes/Transitions.ts

```diff
@@ -105,9 +107,18 @@ export class Transitions {
      * @see https://swup.js.org/hooks/#visit-start
      * @param visit: VisitType
      */
-    onVisitStart() {
+    onVisitStart(visit: VisitType) {
         document.documentElement.classList.add(Transitions.TRANSITION_CLASS);
         document.documentElement.classList.remove(Transitions.READY_CLASS);
+
+        if (visit.a11y) {
+            visit.a11y.focus = 'h1#page-heading';
+
+            // Best not to assume where to move the focus on fragment visits
+            if (visit.fragmentVisit) {
+                visit.a11y.focus = false;
+            }
+        }
```

Comment:
I would discourage this for most standard pages.

The `body` should be the default focused element.

On pages that use modals/drawers, that container or its main heading should receive focus.

If you insist on keeping this, the logic should be collapsed into a single statement:

```suggestion
        // Best not to assume where to move the focus on fragment visits
        if (visit.a11y && !visit.fragmentVisit) {
            visit.a11y.focus = 'h1#page-heading';
        }
```

Reply:
Pages using modals/drawers are typically in a `<dialog>` which automatically focuses its first focusable element upon `showModal()` call.

The issue I see with other types of fragment visits (probably a single section inside the main of the page) resetting the focus to the body is it will require the user to travel again across the page to reach the updated content which could be irritating and potentially disorienting? But at the same time, it's how it would behave without an SPA system... so I'm not entirely sold on that

Also, I posted a [comment](https://github.com/locomotivemtl/astro-boilerplate/pull/34#discussion_r1832853750) asking for you advice regarding this bit of code but github put it as "pending" which prevented you from seeing it.. my bad! It brings some explanations to my thought process there

Reply:
Stick to Web standards and its default, established, behaviours.

If a project's stakeholders can afford to hire accessibility consultants and dedicate time to testing with real users, then the behaviour can be adjusted to accommodate their expectations.